seccomp/src/lib.rs

```diff
@@ -1154,58 +1154,44 @@ mod tests {
     const KVM_GET_PIT2_MSB: u64 = 0x0000_ae9f;
     const KVM_GET_PIT2_LSB: u64 = 0x8070_0000;
 
-    const EXTRA_SYSCALLS: [i64; 5] = [
-        libc::SYS_rt_sigprocmask,
-        libc::SYS_sigaltstack,
-        libc::SYS_munmap,
-        libc::SYS_exit,
-        libc::SYS_rt_sigreturn,
-    ];
-
     fn validate_seccomp_filter(
         rules: Vec<(i64, Vec<SeccompRule>)>,
         validation_fn: fn(),
         should_trigger_sigsys: bool,
     ) {
-        let mut filter =
-            SeccompFilter::new(rules.into_iter().collect(), SeccompAction::Kill).unwrap();
+        let filter = SeccompFilter::new(rules.into_iter().collect(), SeccompAction::Kill).unwrap();
 
-        let triggered_sigsys: Arc<AtomicBool> = Arc::new(AtomicBool::new(true));
+        let triggered_sigsys: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
         let shared_triggered_sigsys = triggered_sigsys.clone();
 
-        // We need 2 threads here: in case of a seccomp denial, the inner thread will be killed
-        // and the outter thread will fail. The execution will be returned to the instruction
-        // that follows the outter thread `join()` method.
-        let outter_thread = thread::spawn(move || {
-            let inner_thread = thread::spawn(move || {
-                // whitelist needed syscalls
-                for syscall in EXTRA_SYSCALLS.iter() {
-                    assert!(filter
-                        .add_rules(
-                            *syscall,
-                            vec![SeccompRule::new(vec![], SeccompAction::Allow)],
-                        )
-                        .is_ok());
-                }
-                // apply filter
+        // We need to execute the seccomp thread inside a catch_unwind block in order to
+        // avoid printing unneeded warnings in case of a seccomp denial.
+        let seccomp_result = panic::catch_unwind(|| {
```

Comment:
While this test is not 100% conclusive (it can, in some cases, provide false negatives) it's still better than nothing.
Unless we find a better solution I say we keep the test and maybe add a comment specifying why this test doesn't fully guarantee SIGSYS correctness.

Better to get some validation than no validation 😄

Reply:
Looks like we aren't the only humans having problem with these kinds of tests: https://github.com/rust-lang/rust/issues/32512

It doesn't look like it's fixed just yet.

Reply:
The logic gets much cleaner if we use `SECCOMP_RET_ERRNO` . Credits to @alexandruag for the idea. 

@andreeaflorescu @acatangiu @sandreim
Please take another look on the changes.

Reply:
I'm learning `seccomp` with these reviews 😄